client/src/main/java/org/eclipse/hono/client/impl/CommandTargetMapperImpl.java

```diff
@@ -85,19 +77,12 @@ public void initialize(final RegistrationClientFactory registrationClientFactory
                 .withTag(TracingHelper.TAG_DEVICE_ID, deviceId)
                 .start();
 
-        return registrationClientFactory.getOrCreateRegistrationClient(tenantId)
-                .compose(client -> client.assertRegistration(deviceId, null, span.context()))
+        return mapperContext.getViaGateways(tenantId, deviceId, span.context())
                 .recover(t -> {
                     LOG.debug("Error getting registration assertion", t);
                     return Future.failedFuture(t);
```

Comment:
The "Error getting registration assertion" log message only looks correct if one knows what `getViaGateways` is doing internally. So, "Error getting 'via' gateways" would be more obvious here.